java/code/src/com/suse/manager/maintenance/IcalUtils.java

```diff
@@ -145,6 +150,25 @@
         return sortedLimited;
     }
 
+    private PeriodList getInitialEvents(Collection<CalendarComponent> events, Period period) {
+        PeriodList periodList = new PeriodList();
+        events.forEach(event -> {
+                    try {
+                        DateTime start = new DateTime(event.getProperty(Property.DTSTART).getValue());
+                        DateTime end = new DateTime(event.getProperty(Property.DTEND).getValue());
+                        // Check if event is contained in given period
+                        if (period.getStart().compareTo(start) <= 0 && period.getEnd().compareTo(end) >= 0) {
```

Comment:
sounds like you want to use `period.includes(start) && period.includes(end)` ?

Reply:
Yeah that's exactly what I wanted here :facepalm:
Over complicating things for no reason it seems :sweat_smile: 
Thanks